### ISSUE TITLE
Regex match validation not applied for child class

### ISSUE THREAD

**Issue by kazper** [JeremySkinner/FluentValidation#312]
I'm trying out FluentValidation for the first time and I encoutered a weird bug. Regex matching does not seem to work for child class. Tried with different regexes and got the same result. I hope I'm wrong somehow.

```
    RuleFor(x => x.PhoneNumber).Matches(@"/.*\S.*/"); // <-- Works.
    RuleFor(x => x.ChildClass.PhoneNumber).Matches(@"/.*\S.*/"); // <-- Does NOT work.
```

**Comment by JeremySkinner** [comment]
There's no reason why it shouldn't work...is the child property definitely
not null?

**Comment by kazper** [comment]
It definitely wasn't null. But in a quick test I couldn't replicate in a dummy project so I probably missed something. I was using a nested class in a partial class if that is of any value...

Now as I've come more accustomed to FluentValidations I refactored some things and use a ChildValidator in the aformentioned code. And it seems to work now. As I couldn't replicate it and it seems to work now I would like to close the issue on behalf of me messing some little detail up.

But I noticed another anomaly regarding the issue - the jQuery _data-val_ attributes are not generated for the child class regex. The regex is applied when form is posted but not client-side. The _data-val_ attributes are generated for other identical field which is not in a child class.
